docs/dictionary.rst

```diff
@@ -0,0 +1,1101 @@
+
+.. updated 2022-05-16  
+
+
+
+.. _dict EC2:
+
+EC2
+---
+*abbreviation for* Elastic Compute Cloud  
+
+	The cloud computing side of :ref:`dict AWS`. When running workflows on these backends, disk size will scale with your workflow requirements automatically. EC2 instances allow you to make use of Amazon's :ref:`dict spot instance` feature, which may reduce the cost of running workflows.  
+
+Further reading: `<https://docs.aws.amazon.com/ec2/index.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict egress:
+
+egress
+------
+pronounced "ee-gress", rhymes with aggress  
+
+	The action of leaving a place. In the context of :ref:`dict cloud computing`, an egress charge is a fee charged for downloading a file. Sometimes, the person hosting the file is charged for data egress. Other times, the person downloading the file is charged.  
+
+.. note:: This term as we define it here is associated with cloud computing and may have different definitions in other contexts.  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict eLwazi:
+
+eLwazi
+------
+pronounced "el-woz-ee", derived from the Xhosa word for knowledge (uLwazi) and the Luganda word for rock symbolizing robustness (Olwazi)  
+
+	An African-lead open data science platform funded as part of the :ref:`dict NIH`'s :ref:`dict DS-I Africa` program. Leverages :ref:`dict Gen3` and :ref:`dict Terra`.  
+
+Further reading: `<https://elwazi.org/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict entry:
+
+entry
+-----
+	A :ref:`dict tool` or :ref:`dict workflow` on Dockstore. A single entry on Dockstore has a description, a link to the original source-control repository, and at least one :ref:`dict descriptor file` which does some sort of computational task using :ref:`dict CWL`, :ref:`dict WDL`, :ref:`dict Nextflow`, or :ref:`dict Galaxy workflow` syntax. An entry can optionally include a :ref:`dict parameter file` that links to open-access test data. A single entry will include all versions of the tool or workflow that has been registered, with that versioning being based upon the versioning and branches of the source-control repository the descriptor file is hosted on (with the exception of a :ref:`dict legacy tool`, which have versioning based upon their Docker image tags), and any version can be pinned as the default. Entries can be added to :ref:`dict collections` associated with a particular :ref:`dict organization`, or added to :ref:`dict categories` so they can be grouped with other entries that have a similar scientific purpose. Entries may also have :ref:`dict labels` attached to them to help them be found via Dockstore's :ref:`dict faceted search` feature. If the entry is registered using the :ref:`dict Dockstore GitHub App`, then the entry will stay in sync automatically with the source-control repository. Additionally, if an entry is a valid :ref:`dict workflow`, any user can use our :ref:`dict launch with` feature to import the workflow to one of our cloud compute partners.  
```

Comment:
I would tend to keep this short and just describe it as shorthand for a tool or workflow. But I'm biased because I don't think we should even surface entry at all, although the cat's out of the bag on that one.

Reply:
I might be misunderstanding what Dockstore envisions an entry to be, but I don't think it's correct to call an entry a shorthand for a tool/workflow any more than it would be say that a GitHub repository is a shorthand for a computer program. An entry can contain a tool/workflow, but it's more than just that.

That said, I am open to keeping this definition in short in favor of https://github.com/dockstore/dockstore/issues/4908 instead.

Reply:
I now better understand our differences on this.

In our Java code for Dockstore we have:

```
abstract class Entry {
...
}

public class Tool extends Entry {
...
}
public class Workflow extends Entry {
...
}
```

So, when I say a workflow is an entry, it literally is that in our class hierarchy behind the scenes. And it's why we use the word entry to refer to a tool or a workflow. It's why `My Entries` on the home page searches your tools and workflows.

But I think you've interpreted it in a different way, where the workflow is the actual executing code, and all the other stuff associated with it is the entry.

While we don't have to echo the class hierarchy in our public facing view, we have so far. As you suggest,  we can revisit with dockstore/dockstore#4908.

Reply:
I still think entry should be defined to as shorthand for tool or workflow. If we want to broaden the definition, we should discuss later as you suggested. As far as I know, we don't refer to entry any other way.